### PR TITLE
Simple/naive example of FFNN using Torch

### PR DESCRIPTION
**Pull request by guillaumevernieres**
#### What was done
- The README is now amazing!!!
- Addition of a simplistic FFNN using Torch, eckit and oops, mostly there to test the build system
- A ctest section that trains the ffnn and does 1 dummy prediction
 
#### What's next
Plenty of direction, from doing actual science to writing a generic class that quite a few application would fit within.

### CODE REVIEW

**Review by RussTreadon-NOAA — APPROVED** [review]
This is neat!   To thoroughly review I should clone, install, and run the ctest.  I have not done this yet.  Doing so would be more a learning exercise for me than a review of the proposed changes.  Given this, my review is limited, at present, to looking at the files to be committed.  They look OK to me.

Approve.

File: sshu/daml_sshu.cc [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#include "eckit/config/YAMLConfiguration.h"
+#include "./eckit/filesystem/PathName.h"
```
Comment (CoryMartin-NOAA) [inline sshu/daml_sshu.cc]: why is one `eckit` and the other `./eckit`?

File: sshu/daml_sshu.cc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+// Define the Feed Forward Neural Net model.
+struct Net : torch::nn::Module {
+  Net(int inputSize, int hiddenSize, int outputSize) {
+    std::cout << inputSize << outputSize << hiddenSize << std::endl;
+    // Define the layers.
+    fc1 = register_module("fc1", torch::nn::Linear(inputSize, hiddenSize));
+    fc2 = register_module("fc2", torch::nn::Linear(hiddenSize, outputSize));
+  }
+
+  // Implement the forward pass.
```
Comment (CoryMartin-NOAA) [inline sshu/daml_sshu.cc:17]: I thought that's been in football for over a century now?!? :-)

File: test/daml_sshu.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+ffnn:
+  inputSize: 5
+  outputSize: 1
+  hiddenSize: 50
+
+training data:
+  dataSize: 2000
+
+training:
+  epochs: 800
```
Comment (CoryMartin-NOAA) [inline test/daml_sshu.yaml]: No EOL

**Review by CoryMartin-NOAA — COMMENTED** [review]
One quick question, one dumb joke, one actual tiny issue

File: sshu/daml_sshu.cc (hunk 1 above)
Comment (guillaumevernieres) [reply sshu/daml_sshu.cc]: ha! good catch, not sure why this even works!

File: sshu/daml_sshu.cc (hunk 2 above)
Comment (guillaumevernieres) [reply sshu/daml_sshu.cc:17]: Have you ever seen that error:
```
[Rank 1] Error: PE is offside.
```
It happens during deep learning when some PE's become self aware and start cheating. That's why we need to quickly switch to gpu's. I think ...

Comment (CoryMartin-NOAA) [reply sshu/daml_sshu.cc:17]: Maybe we just need to increase the weight of the penalty

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**